src/averageAges.js

```diff
@@ -15,11 +15,12 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menArr = people.filter(x => x.sex === 'm'
+    && (!century || Math.ceil(x.died / 100) === century));
+
+  const ages = menArr.reduce((acc, x) => acc + x.died - x.born, 0);
+
+  return ages / menArr.length;
```

Comment:
you duplicate getting average ages logic in the next function. move it to a separate service function and reuse